crucible-llvm/src/Lang/Crucible/LLVM/Intrinsics/Libc.hs

```diff
@@ -495,6 +523,29 @@ callRealloc sym mvar alignment (regValue -> ptr) (regValue -> sz) =
        ]
 
 
+callPosixMemalign
+  :: (IsSymInterface sym, HasPtrWidth wptr, wptr ~ ArchWidth arch, ?lc :: TypeContext)
+  => sym
+  -> GlobalVar Mem
+  -> RegEntry sym (LLVMPointerType wptr)
+  -> RegEntry sym (BVType wptr)
+  -> RegEntry sym (BVType wptr)
+  -> OverrideSim p sym (LLVM arch) r args ret (RegValue sym (BVType 32))
+callPosixMemalign sym mvar (regValue -> outPtr) (regValue -> align) (regValue -> sz) =
+  case asUnsignedBV align >>= toAlignment . toBytes of
+    Nothing -> fail $ unwords ["posix_memalign: invalid alignment value:", show (printSymExpr align)]
+    Just a ->
```

Comment:
This could be replaced by something like:
```haskell
p <- callMalloc sym mvar alignment sz
mem <- readGlobal mvar
writeGlobal mvar =<< storeRaw sym mem' outPtr (bitvectorType (dl^.ptrSize)) (dl^.ptrAlign) (ptrToPtrVal p)
```
Both this and the implementation of `callMalloc` make me feel like we need a `modifyGlobal` combinator for the `readGlobal`/`writeGlobal` pairing.

Reply:
Yeah, a `modifyGlobal` combinator is a good idea.